molgenis-data-excel/src/main/java/org/molgenis/data/excel/xlsx/XlsxWriter.java

```diff
@@ -0,0 +1,175 @@
+package org.molgenis.data.excel.xlsx;
+
+import static java.util.Objects.requireNonNull;
+import static jdk.nashorn.internal.runtime.JSType.toLong;
+import static org.molgenis.data.DataConverter.toBoolean;
+import static org.molgenis.data.DataConverter.toDouble;
+import static org.molgenis.data.DataConverter.toInt;
+import static org.molgenis.data.DataConverter.toLocalDate;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.Date;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.molgenis.data.excel.xlsx.exception.MaximumSheetNameLengthExceededException;
+import org.molgenis.data.excel.xlsx.exception.UnsupportedValueException;
+import org.molgenis.data.excel.xlsx.exception.XlsxWriterException;
+
+public class XlsxWriter implements AutoCloseable {
```

Comment:
Besides the Sonarlint issues mentioned in commands below IntelliJ reports 12 warnings that should be fixed.